### PR TITLE
Feature/add rfc4122 uuids

### PR DESCRIPTION
**Pull request by k14uz**
The modifications in this pull request are as follows:

- ~~**Integration of UUID Package**: The UUID package has been integrated to facilitate the generation of unique identifiers~~. (`Nanoid` module is being used from existing packages)
- **Assignments in Core Modules**: The UUID functionality has been assigned in the following modules:
    - `packages/core/svelte/use-chat.ts`
    - `packages/core/vue/use-chat.ts`
    - `packages/core/vue/use-completion.ts`
    - `packages/core/svelte/use-completion.ts`

This modification ensures the `uniqueId` is in compliance with [RFC4122](https://www.ietf.org/rfc/rfc4122.txt).

### CODE REVIEW

**Comment by jaredpalmer** [comment]
Can we use nanoid to produce the same result? We already use that package.

**Comment by sirlolcat** [comment]
> Can we use nanoid to produce the same result? We already use that package.

I adjusted it accordingly, let me know if there is anything you'd like me to address.

File: packages/core/streams/ai-stream.ts [hunk 1]
```diff
@@ -5,110 +5,142 @@ import {
...
-    })
+  const responseBodyStream = response.body || createEmptyReadableStream()
 
-  return stream
+  return responseBodyStream
     .pipeThrough(createEventStreamTransformer(customParser))
     .pipeThrough(createCallbacksTransformer(callbacks))
 }
+
+/**
```
Comment (jaredpalmer) [inline packages/core/streams/ai-stream.ts:136]: Remove changes to ai-stream, seems like mixing PRs here.

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: packages/core/streams/ai-stream.ts (hunk 1 above)
Comment (sirlolcat) [reply packages/core/streams/ai-stream.ts]: My apologies, there was a mix-up in the local setup. I reverted it now.

File: packages/core/svelte/use-chat.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export function useChat({
   body
 }: UseChatOptions = {}): UseChatHelpers {
   // Generate a unique ID for the chat if not provided.
-  const chatId = id || `chat-${uniqueId++}`
+  const chatId = id || `chat-${uniqueId}`
```
Comment (jaredpalmer) [inline packages/core/svelte/use-chat.ts:63]: this isn't calling the function, it's just reusing the same one.

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: packages/core/svelte/use-chat.ts (hunk 2 above)
Comment (sirlolcat) [reply packages/core/svelte/use-chat.ts:63]: Great catch! I'll fix it now.

**Review by MaxLeiter — CHANGES_REQUESTED** [review]
Can you modify the react hooks to also use `nanoid`? [`useId`](https://react.dev/reference/react/useId) should be used for accessibility and not general purpose IDs.

**Comment by jaredpalmer** [comment]
Let's update and merge.

**Review by MaxLeiter — APPROVED** [review]

**Pull request merged**